sources/views/activitiesForm.js

```diff
@@ -43,15 +43,19 @@ export default class ActivitiesForm extends JetView {
 		this.form = this.$$("activitiesForm");
 	}
 
-	showForm(id) {
-		if (id && activities.exists(id)) {
+	showForm(id, add) {
```

Comment:
"id" and "add" parameters do the same logic, if id exists - it is edit, if not it is add, the same works with add, you should leave only one param

Reply:
also you already checked if (id) { ... in previous call, so id will exists

Reply:
There is another logic. We call "showForm" method from "Contacts" page and from "Activities" page. If that method was called from "Activity" page it'll open an empty form to add new activity. But if it was called from "Contacts" page, it gets additional "add" parameter, that will open form with preselected contact that can't be changed. I can leave only "id" check,  but contacts and activity collections can have the same "id" property. Alternatively, I can separate the functions, but they would almost repeat each other. What is the right desigion?

Reply:
showForm(activityId, contactId) {
		if (activityId && activities.exists(activityId)) {
			const item = activities.getItem(activityId);
			this.form.setValues(item);
		}
		if (contactId && contacts.exists(contactId)) {
			this.$$("combo").setValue(contactId);
			this.$$("combo").define("readonly", true);
		}

		this.getRoot().show();

		const action = !activityId ? "Add" : "Edit";
		this.$$("header").setValues({action});
		this.$$("button").setValue(action);
	}

Reply:
and pass activityId and contactId where any of param is nesessary

params id and add are confusing anyone who see your code, it is complicated to understand. Now,  If you pass activityId it means you edit activity, if not - add activity. And if you pass contactId it means it activity for the contact

Reply:
It's really looks much better and easier to understand. Thank you.